pkg/credentials/secret.go

```diff
@@ -75,14 +77,21 @@ func Ensure(s *state.State) error {
 
 	s.Logger.Infoln("Creating machine-controller credentials secret...")
 
-	mcCreds, err := ProviderCredentials(s.Cluster.CloudProvider, s.CredentialsFilePath, TypeMC)
+	providerCreds, err := ProviderCredentials(s.Cluster.CloudProvider, s.CredentialsFilePath, TypeMC)
 	if err != nil {
 		return errors.Wrap(err, "unable to fetch cloud provider credentials")
 	}
 
-	mcSecret := credentialsSecret(SecretNameMC, mcCreds)
+	mcSecret := credentialsSecret(SecretNameMC, providerCreds)
 	if createErr := clientutil.CreateOrReplace(context.Background(), s.DynamicClient, mcSecret); createErr != nil {
-		return errors.Wrap(createErr, "failed to ensure credentials secret")
+		return errors.Wrap(createErr, "failed to ensure credentials secret for machine-controller")
+	}
+
+	if s.Cluster.AddonOperatingSystemManagerEnabled() {
+		osmSecret := credentialsSecret(SecretNameOSM, providerCreds)
+		if createErr := clientutil.CreateOrReplace(context.Background(), s.DynamicClient, osmSecret); createErr != nil {
+			return errors.Wrap(createErr, "failed to ensure credentials secret for operating-system-manager")
+		}
 	}
```

Comment:
Why do we need a dedicated secret for OSM? Can we just reuse the MC secret?

Reply:
We can but I'm not really a big fan of sharing secrets between distinct apps, even if they are exactly the same. But yes we can probably do that in this case, should I?